src/openms/source/PROCESSING/SMOOTHING/GaussFilter.cpp

```diff
@@ -140,6 +141,57 @@ namespace OpenMS
     }
   }
 
+  void GaussFilter::filter(Mobilogram & mobilogram)
+  {
+    if (param_.getValue("use_ppm_tolerance").toBool())
+    {
+      throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, 
+        "GaussFilter: Cannot use ppm tolerance on mobilogram");
+    }
+
+    bool found_signal = false;
+    const Size data_size = mobilogram.size();
+    std::vector<double> im_in(data_size), int_in(data_size), im_out(data_size), int_out(data_size);
+
+    // copy spectrum to container
+    for (Size p = 0; p < mobilogram.size(); ++p)
+    {
+      im_in[p] = mobilogram[p].getMobility();
+      int_in[p] = mobilogram[p].getIntensity();
+    }
+
+    // apply filter
```

Comment:
can you check i we could use .getPos() so we can reuse the different implementations (or parts of it) for spectra, chromatograms, and mobilograms

Reply:
nevermind... I think it is fine here... it mainly does copying